### PR TITLE
Chore: Update  dependencies and remove unneeded ones

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
Update `go.mod` dependencies.

**Release note**:
```release-note
Updated dependencies.
```

### CODE REVIEW

**Review by lyarwood — COMMENTED** [review]
LGTM but I posted https://github.com/kubevirt/ssp-operator/pull/505 last week and was getting some odd failures with ginkgo `v2.8.3`, can't see the same here so hopefully the other dep bumps have resolved whatever it was I was hitting. I'll wait for CI before approving.

**Comment by 0xFelix** [comment]
@lyarwood What about the changes introduced in https://github.com/kubevirt/ssp-operator/pull/505/commits/b1f9f04e32ac1076e09c7be559f2088efdcac748?

**Comment by lyarwood** [comment]
> @lyarwood What about the changes introduced in [b1f9f04](https://github.com/kubevirt/ssp-operator/commit/b1f9f04e32ac1076e09c7be559f2088efdcac748)?

I'll handle that in a new PR once this lands.

File: internal/template-validator/webhooks/hook.go [hunk 1]
```diff
@@ -154,11 +162,19 @@ func serve(resp http.ResponseWriter, req *http.Request, admit admitFunc) {
 		return
 	}
 
-	log.Log.V(8).Infof("admission review:\n%s", spew.Sdump(review))
+	if reviewJson, err := json.Marshal(review); err == nil {
+		log.Log.V(8).Infof("admission review:\n%s", string(reviewJson))
+	} else {
+		log.Log.V(8).Infof("admission review:\nCould not marshall review to json: %s", err.Error())
```
Comment (0xFelix) [inline internal/template-validator/webhooks/hook.go]: Typo: marshal / marshall

File: internal/template-validator/webhooks/hook.go [hunk 2]
```diff
@@ -89,8 +88,17 @@ func (w *webhooks) admitVm(ar *admissionv1.AdmissionReview) *admissionv1.Admissi
...
+	if vmJson, err := json.Marshal(vm); err == nil {
+		log.Log.V(8).Infof("admission vm:\n%s", string(vmJson))
+	} else {
+		log.Log.V(8).Infof("admission vm:\nCould not marshall VM to json: %s", err.Error())
+	}
+
+	if rulesJson, err := json.Marshal(rules); err == nil {
+		log.Log.V(8).Infof("admission rules:\n%s", string(rulesJson))
+	} else {
+		log.Log.V(8).Infof("admission vm:\nCould not marshall rules to json: %s", err.Error())
```
Comment (0xFelix) [inline internal/template-validator/webhooks/hook.go]: Typo: marshal / marshall

File: internal/template-validator/webhooks/hook.go [hunk 3]
```diff
@@ -89,8 +88,17 @@ func (w *webhooks) admitVm(ar *admissionv1.AdmissionReview) *admissionv1.Admissi
 		return ToAdmissionResponseError(err)
 	}
 
-	log.Log.V(8).Infof("admission vm:\n%s", spew.Sdump(vm))
-	log.Log.V(8).Infof("admission rules:\n%s", spew.Sdump(rules))
+	if vmJson, err := json.Marshal(vm); err == nil {
+		log.Log.V(8).Infof("admission vm:\n%s", string(vmJson))
+	} else {
+		log.Log.V(8).Infof("admission vm:\nCould not marshall VM to json: %s", err.Error())
```
Comment (0xFelix) [inline internal/template-validator/webhooks/hook.go]: Typo: marshal / marshall

File: internal/template-validator/webhooks/hook.go [hunk 4]
```diff
@@ -154,11 +162,19 @@ func serve(resp http.ResponseWriter, req *http.Request, admit admitFunc) {
...
+		log.Log.V(8).Infof("admission review:\nCould not marshall review to json: %s", err.Error())
+	}
 
 	reviewResponse := admit(review)
 
-	log.Log.V(8).Infof("admission review response:\n%s", spew.Sdump(reviewResponse))
+	if reviewResponseJson, err := json.Marshal(reviewResponse); err == nil {
+		log.Log.V(8).Infof("admission review response:\n%s", string(reviewResponseJson))
+	} else {
+		log.Log.V(8).Infof("admission review response:\nCould not marshall reviewResponse to json: %s", err.Error())
```
Comment (0xFelix) [inline internal/template-validator/webhooks/hook.go]: Typo: marshal / marshall

File: internal/template-validator/webhooks/hook.go (hunk 1 above)
Comment (0xFelix) [reply internal/template-validator/webhooks/hook.go]: Can you dump out the review with `%v`?

File: internal/template-validator/tlsinfo/tlsinfo.go [hunk 5]
```diff
@@ -135,15 +136,16 @@ func notify(channel chan struct{}) {
 func (ti *TLSInfo) updateCertificates(directory string) error {
 	cert, err := loadCertificates(directory)
 	if err != nil {
-		log.Log.Reason(err).Infof("failed to load the certificate in %s", directory)
+		logger.Log.Error(err, "failed to load certificates",
```
Comment (0xFelix) [inline internal/template-validator/tlsinfo/tlsinfo.go:139]: Changing from Info to Error?

File: internal/template-validator/webhooks/hook.go (hunk 1 above)
Comment (akrejcir) [reply internal/template-validator/webhooks/hook.go]: I think json is easier to read. `%v` is more verbose, I think it prints types of fields and addresses of pointers.

File: internal/template-validator/webhooks/hook.go (hunk 4 above)
Comment (akrejcir) [reply internal/template-validator/webhooks/hook.go]: done

File: internal/template-validator/webhooks/hook.go (hunk 3 above)
Comment (akrejcir) [reply internal/template-validator/webhooks/hook.go]: Done

File: internal/template-validator/webhooks/hook.go (hunk 2 above)
Comment (akrejcir) [reply internal/template-validator/webhooks/hook.go]: Done

**Pull request merged**